napari/components/layerlist.py

```diff
@@ -108,71 +130,20 @@ def unselect_all(self, ignore=None):
         ignore : Layer | None
             Layer that should not be unselected if specified.
         """
-        for layer in self:
-            if layer.selected and layer != ignore:
-                layer.selected = False
-
-    def select_all(self):
-        """Selects all layers."""
-        for layer in self:
-            if not layer.selected:
-                layer.selected = True
-
-    def remove_selected(self):
-        """Removes selected items from list."""
-        to_delete = []
-        for i in range(len(self)):
-            if self[i].selected:
-                to_delete.append(i)
-        to_delete.reverse()
-        for i in to_delete:
-            self.pop(i)
-        if len(to_delete) > 0:
-            first_to_delete = to_delete[-1]
-            if first_to_delete == 0 and len(self) > 0:
-                self[0].selected = True
-            elif first_to_delete > 0:
-                self[first_to_delete - 1].selected = True
-
-    def select_next(self, shift=False):
-        """Selects next item from list."""
-        selected = []
-        for i in range(len(self)):
-            if self[i].selected:
-                selected.append(i)
-        if len(selected) > 0:
-            if selected[-1] == len(self) - 1:
-                if shift is False:
-                    self.unselect_all(ignore=self[selected[-1]])
-            elif selected[-1] < len(self) - 1:
-                if shift is False:
-                    self.unselect_all(ignore=self[selected[-1] + 1])
-                self[selected[-1] + 1].selected = True
-        elif len(self) > 0:
-            self[-1].selected = True
-
-    def select_previous(self, shift=False):
-        """Selects previous item from list."""
-        selected = []
-        for i in range(len(self)):
-            if self[i].selected:
-                selected.append(i)
-        if len(selected) > 0:
-            if selected[0] == 0:
-                if shift is False:
-                    self.unselect_all(ignore=self[0])
-            elif selected[0] > 0:
-                if shift is False:
-                    self.unselect_all(ignore=self[selected[0] - 1])
-                self[selected[0] - 1].selected = True
-        elif len(self) > 0:
-            self[0].selected = True
+        warnings.warn(
+            "'layers.unselect_all()' is deprecated and will be removed in "
+            ">=v0.4.9. Please use 'layers.selection.clear()'.  To unselect "
+            "everything but a set of ignored layers, use "
+            r"'layers.selection.intersection_update({ignored})'",
+            category=DeprecationWarning,
+            stacklevel=2,
```

Comment:
```suggestion
            "'layer.unselect_all()' is deprecated and will be removed in "
            "v0.4.9 or later. Please use 'viewer.layers.selection.clear()'."
            " To unselect everything except a set of ignored layers, use "
            r"'viewer.layers.selection.intersection_update({ignored})'",
            category=FutureWarning,
            stacklevel=2,
```

@tlambert03 I don't understand the reason for the second part of the message — that is new functionality, right? And if so, shouldn't it go in some documentation rather than a deprecation message?

Reply:
Update: never mind, found the method signature. LOL